frontend/src/components/App.tsx

```diff
@@ -0,0 +1,116 @@
+import { Input, Button, CircularProgress } from '@mui/material';
+
+function App() {
+  const appContext = useContext(AppContext);
+  const [practitioners, setPractitioners] = useState<NormalizedPractitioner[]>([]);
+  const [searchTerm, setSearchTerm] = useState('');
+
+  useEffect( () => {
+    let searchUrl = HapiPractitionersFetchRequestUrl;
+
+    if (!appContext?.state.fetched || searchTerm.length === 0) {
+      axios.get<HapiPractitionersResponse>(searchUrl)
+        .then((res: AxiosResponse<HapiPractitionersResponse>) => {
+          if (res.data.entry) {
+            appContext?.dispatch({
+              type: 'set-practitioners',
+              payload: res.data
+            });
+          }
+        })
+        .catch((err) => console.error(err));
+    } else {
+      setPractitioners(appContext.state.practitioners);
+    }
+
+    if (searchTerm.length > 0) {
+      searchUrl = `${HapiPractitionersFetchRequestUrl}/?given=${searchTerm}&_format=json&_pretty=true`;
+
+      axios.get<HapiPractitionersResponse>(searchUrl)
+        .then((res: AxiosResponse<HapiPractitionersResponse>) => {
+          if (res.data.entry) {
+            appContext?.dispatch({
+              type: 'set-practitioners',
+              payload: res.data
+            });
+          }
+        })
+        .catch((err) => console.error(err));
+    }
+  }, [appContext?.state.fetched, searchTerm]);
+
+  const getPreviousPage = () => {
+    axios.get<HapiPractitionersResponse>(appContext?.state.previousPage as string)
+      .then((res: AxiosResponse<HapiPractitionersResponse>) => {
+        if (res.data.entry) {
+          appContext?.dispatch({
+            type: 'set-practitioners',
+            payload: res.data
+          });
+        }
+      })
+      .catch((err) => console.error(err));
+  }
+
+  const getNextPage = () => {
+    axios.get<HapiPractitionersResponse>(appContext?.state.nextPage as string)
+      .then((res: AxiosResponse<HapiPractitionersResponse>) => {
+        if (res.data.entry) {
+          appContext?.dispatch({
+            type: 'set-practitioners',
+            payload: res.data
+          });
+        }
+      })
+      .catch((err) => console.error(err));
+  }
+
+  return (
+    <div className="text-center">
+      <header className="min-h-2 p-5 bg-sky-700 text-white font-bold pos drop-shadow-md">
+        <nav>
+          <div className="container flex flex-wrap items-center justify-between mx-auto">
+            <h1>Frontend Assessment</h1>
+            <Input
+              classes={{ inputTypeSearch: 'search'}}
+              onChange={(ev) => setSearchTerm(ev.target.value)}
+            />
+          </div>
+        </nav>
+      </header>
+      <main className="bg-sky-900 text-white min-h-screen flex flex-col items-center justify-center content-center p-5">
+        {
+          practitioners.length ?
+            <PractitionerFeed
+                practitioners={practitioners}
+            />
+          : <CircularProgress />
```

Comment:
Doesn't this potentially result in an endless spinner if the search term is not found? I have not yet run the app, so not sure, but that seems likely.

A better approach may be to have a separate UI state for a "not found" status. That would give the user better feedback.

Reply:
you're right. there's no time out to display no results. i was just doing something quick and didn't think too much about it so that's an oversight of mine. i also wanted to add an error handler too but that would probably be the next commit i'd do. your approach sounds like a good idea for sure!